absl/strings/escaping_test.cc

```diff
@@ -635,4 +635,99 @@ TEST(HexAndBack, HexStringToBytes_and_BytesToHexString) {
   EXPECT_EQ(hex_only_lower, hex_result);
 }
 
+TEST(NonNullTerminatedStrings, OctalConversion) {
+  absl::string_view reference("\\111", 2);
+  std::string expected = "\1";
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+
+  EXPECT_EQ(unescaped_str, expected);
+}
+
+TEST(NonNullTerminatedStrings, InvalidHexConversion) {
+  absl::string_view reference("\\x1", 2);
+
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape(reference, &unescaped_str));
+}
+
+TEST(NonNullTerminatedStrings, HexConversion) {
+  absl::string_view reference("\\x11", 3);
+  std::string expected = "\x1";
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+  EXPECT_EQ(unescaped_str, expected);
+}
+
+TEST(NonNullTerminatedStrings, UtfConversion_u) {
+  absl::string_view invalid("\\u11111", 5);
+
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape(invalid, &unescaped_str));
+}
+
+TEST(NonNullTerminatedStrings, UtfConversion_U) {
+  absl::string_view invalid("\\U0000000000000", 9);
+
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape(invalid, &unescaped_str));
+}
+
+TEST(NonNullTerminatedStrings, InvalidOctal) {
+  absl::string_view reference("\\1z1", 3);
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+
+  EXPECT_EQ(unescaped_str, "\1z");
+}
+
+TEST(NonNullTerminatedStrings, InvalidHex) {
+  absl::string_view reference("\\xz", 3);
```

Comment:
It's not obvious to me why you're specifying the size of the `string_view` here.  Wouldn't `std::string reference = "\\xz";` have the same behavior?

Reply:
You are right. I thought I needed to adjust the size to get the right behavior, and ended up with the full string without realizing it :)

FYI, this test is designed to catch this error:

```
        case 'x':                                                                                                       
        case 'X': {                                                                                                     
          if (p >= last_byte) {                                                                                         
            if (error) *error = "String cannot end with \\x";                                                           
            return false;                                                                                               
          } else if (!absl::ascii_isxdigit(p[1])) {                                                                     
            if (error) *error = "\\x cannot be followed by a non-hex digit";                                            
            //return false;      <---- This is the only test failing if this return statement is missing                                                                                                                                                  
          }    
```